kasa/smartdevice.py

```diff
@@ -191,20 +192,18 @@ def __init__(
         self,
         host: str,
         *,
-        port: Optional[int] = None,
-        credentials: Optional[Credentials] = None,
-        timeout: Optional[int] = None,
+        config: Optional[DeviceConfig] = None,
+        protocol: Optional[TPLinkProtocol] = None,
```

Comment:
I suppose not having `protocol` but force to pass it inside the config would make using the API too messy? Otherwise that would be the preferred way, right?

Reply:
Because discovery currently constructs the `protocol` which holds it's copy of the config it's cleaner to be able to pass it without the config.  An alternative could be to have the `SmartDevice.init()` inline import `get_protocol` and construct the protocol internally.  I could look at that for a separate PR as I think it would work off the top of my head but there may be some circular issues.

Reply:
Ok, let's keep it for the time being and get this merged, we can revisit it if necessary.